libs/stats/odc/stats/_fc_percentiles.py

```diff
@@ -0,0 +1,93 @@
+"""
+Fractional Cover Percentiles
+"""
+from typing import Optional, Tuple
+from itertools import product
+import dask.array as da
+import xarray as xr
+from odc.stats.model import Task
+from odc.algo.io import load_with_native_transform
+from odc.algo import keep_good_only, apply_numexpr
+from odc.algo._percentile import xr_percentile
+from odc.algo._masking import _first_fuser
+from .model import StatsPluginInterface
+from . import _plugins
+
+
+class StatsFCP(StatsPluginInterface):
+    
+    NAME = "ga_fc_percentiles"
+    SHORT_NAME = NAME
+    VERSION = "0.0.1"
+    PRODUCT_FAMILY = "fc_percentiles"
+
+    def __init__(
+        self,
+        resampling: str = "bilinear",
+    ):
+        self.resampling = resampling
+
+    @property
+    def measurements(self) -> Tuple[str, ...]:
+        _measurments = [f"{b}_pc_{p}" for b, p in product(["pv", "bs", "npv", "ue"], ["10", "50", "90"])]
+        _measurments.append("is_ever_wet")
+        return _measurments
+
+    @staticmethod
+    def _native_tr(xx):
+        """
+        Loads in the data in the native projection. It performs the following:
+
+        1. Loads all the fc and WOfS bands
+        2. Extracts the clear dry and clear wet flags from WOfS
+        3. Drops the WOfS band
+        4. Stores the clear wet flags for QA later on 
+        5. Masks out all pixels that are not clear and dry to a nodata value of 255
+        6. Discards the clear dry flags
+        """
+
+        # set terrain flag to zero
+        water = da.bitwise_and(xx["water"], 0b11101111)
+        xx = xx.drop_vars(["water"])
+
+        # equivalent to water & 0b11101111
+        expression = "((water << 4) >> 4) + ((water >> 5) << 5)"
+        water = apply_numexpr(expression, xx, dtype="bool")
```

Comment:
but you already dropped `water` for `xx` above?

Reply:
oh woops I pushed the wrong thing